setup.py

```diff
@@ -1,27 +1,167 @@
-from setuptools import setup, find_packages
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+This file is part of nucypher.
 
-VERSION = '0.1'
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
 
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
 
-# NOTE: Use Pipfile & Pipfile.lock to manage dependencies\
-INSTALL_REQUIRES = []
-TESTS_REQUIRE = []
+You should have received a copy of the GNU General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import os
+import sys
+
+from setuptools import setup
+from setuptools.command.install import install
+
+
+#
+# Metadata
+#
+
+PACKAGE_NAME = 'nucypher'
+BASE_DIR = os.path.dirname(__file__)
+
+ABOUT = dict()
+with open(os.path.join(BASE_DIR, PACKAGE_NAME, "__about__.py")) as f:
+    exec(f.read(), ABOUT)
+
+with open(os.path.join(BASE_DIR, "README.md")) as f:
+    long_description = f.read()
+
+
+#
+# Utilities
+#
+
+class VerifyVersionCommand(install):
+    """Custom command to verify that the git tag matches our version"""
+    description = 'verify that the git tag matches our version'
+
+    def run(self):
+        tag = os.getenv('CIRCLE_TAG')
+        if tag.startswith('v'):
+            tag = tag[1:]
+
+        version = ABOUT['__version__']
+        if version.startswith('v'):
+            version = version[1:]
+
+        if tag != version:
+            info = "Git tag: {0} does not match the version of this app: {1}".format(
+                os.getenv('CIRCLE_TAG'), ABOUT['__version__']
+            )
+            sys.exit(info)
+
+
+#
+#  Dependencies
+#
+
+INSTALL_REQUIRES = [
+
+    # NuCypher
+    'umbral',
+    'constant-sorrow',
+    'bytestringSplitter',
+    'hendrix>=3.1.0',
+
+    # Third Party (General)
+    'cryptography>=2.3',
+    'pysha3',
+    'requests',
+    'sqlalchemy',
+    'apistar==0.5.42',
+    'tzlocal==1.5.1',
+    'maya',
+
+    # Third Party (Ethereum)
+    'coincurve>=8.0.2',
+    'eth-utils',
+    'eth-keys',
+    'eth-tester>=0.1.0b33',
+    'py-evm>=0.2.0a33',
+    'py-solc',
+    'web3',
+
+    # Third Party (Configuration + CLI)
+    'appdirs',
+    'click',
+    'colorama',
+    'sentry-sdk'
+]
+
+TESTS_REQUIRE = [
+    'pytest',
+    'pytest-mypy',
+    'pytest-twisted',
+    'pytest-cov',
+    'mypy',
+    'codecov',
+    'coverage',
+    'moto'
+]
+
+DEPLOY_REQUIRES = [
+    'bumpversion',
+    'boto3',
```

Comment:
When is `boto3` used, for Ursulas or for Alices also?

Currently, if we `pip install` nucypher, `boto3` is not installed unless it's `deployment`, and anything from `storages` module is unusable (becase `boto3` is imported there)

Reply:
Good catch.  We use `boto3` for ansible provisioning, as well as the S3 Storage backend adaptor.  I can move it over to `INSTALL_REQUIRES`.

Reply:
:+1: